### PR TITLE
LPS-112988 Replace Global Menu markup and styles and refactor React component

### PR DESCRIPTION
**Pull request by carloslancha**
As a first step to implement the new Global Menu (some kind of mega dropdown that holds all the items present in the Product Menu that are not specific to the context of the page) Echo Team developed a PoC creating a react component that uses ClayModal to show this new feature.

The purpose of this pr is to replace the markup and some styling for that Global Menu to fit with UX design, remove the use of ClayModal and show it as a mega dropdown, plus optimizing and improving the existing React component.

At this moment, some work needs to be done and merged by Echo Team in order to see any content on the global menu, so you won't be able to watch it in action propertly.

Anyway, the steps to reproduce it, getting an empty global menu, are:

- Deploy this pr
- Deploy the [config file](https://issues.liferay.com/secure/attachment/356992/com.liferay.product.navigation.global.menu.web.internal.configuration.FFProductNavigationGlobalMenuConfiguration.config)
- Go to any site and click the Grid Icon on the right side of the top bar.

This is how it would look with some elements:

image

Previously: https://github.com/wincent/liferay-portal/pull/281

### CODE REVIEW

**Comment by wincent** [comment]
Just leaving a note here for myself to find previous PR: https://github.com/wincent/liferay-portal/pull/281

**Comment by markocikos** [comment]
@carloslancha To avoid conflicts, I'll resend https://github.com/wincent/liferay-portal/pull/285 with no changes on the global menu. Could you please add [that change](https://github.com/wincent/liferay-portal/pull/285/commits/c101f9ff713a21bf35633a8221bc5d47b00f780f#diff-85d5943f016624b8ad04245c8b434faeR122-R131) to your PR.

**Comment by carloslancha** [comment]
Sure @markocikos thx!

**Comment by carloslancha** [comment]
ci:test:sf

**Comment by wincent** [comment]
ci:test:relevant

**Comment by wincent** [comment]
ci:test:bundle

File: modules/apps/product-navigation/product-navigation-global-menu-web/src/main/resources/META-INF/resources/js/GlobalMenu.js [hunk 1]
```diff
@@ -13,188 +13,207 @@
...
+													className="dropdown-item"
+													href={url}
+												>
+													<ClaySticker
+														className="inline-item-before"
+														inline={true}
+														size="sm"
+													>
+														<img
+															className="sticker-img"
```
Comment (wincent) [inline modules/apps/product-navigation/product-navigation-global-menu-web/src/main/resources/META-INF/resources/js/GlobalMenu.js:104]: Should this `img` have an empty `alt` for accessibility? Seems "[decorative](https://www.w3.org/WAI/tutorials/images/decorative/)" to me, so I'd expect that we'd want the screen reader to skip mention of it.

Comment (carloslancha) [reply modules/apps/product-navigation/product-navigation-global-menu-web/src/main/resources/META-INF/resources/js/GlobalMenu.js:104]: @marcoscv-work something to say about it?

**Comment by wincent** [comment]
This is what I see when I run the bundle with the config:

Screenshot 2020-05-22 -141127-FuGiBzKO@2x

No errors in the console though... 🤔

**Comment by carloslancha** [comment]
@wincent exactly, as said in the description there's work that need to be done by Echo Team in order to show elements in the Global Menu, so for now you only can see an empty box 🤷

**Comment by wincent** [comment]
Ah. Got confused by the screenshot. Thanks for clarifying.

**Comment by wincent** [comment]
I'm going to disconnect now. Leaving it in the CI's hands for now — if it wants to forward, it will forward.

**Comment by wincent** [comment]
ci:forward

**Comment by jbalsas** [comment]
believeinyourdreams

**Comment by carloslancha** [comment]
image

**Comment by wincent** [comment]
deadly-serious

**Pull request closed without merging**